src/kem/kyber/serialize.rs

```diff
@@ -89,16 +77,18 @@ pub(super) fn deserialize_to_uncompressed_ring_element(
 }
 
 #[inline(always)]
-fn compress_then_serialize_10<const OUT_LEN: usize>(
-    re: KyberPolynomialRingElement,
-) -> [u8; OUT_LEN] {
+fn compress_then_serialize_10<const OUT_LEN: usize>(re: PolynomialRingElement) -> [u8; OUT_LEN] {
     let mut serialized = [0u8; OUT_LEN];
 
     for (i, coefficients) in re.coefficients.chunks_exact(4).enumerate() {
-        let coefficient1 = compress_q(10, to_unsigned_representative(coefficients[0]));
-        let coefficient2 = compress_q(10, to_unsigned_representative(coefficients[1]));
-        let coefficient3 = compress_q(10, to_unsigned_representative(coefficients[2]));
-        let coefficient4 = compress_q(10, to_unsigned_representative(coefficients[3]));
+        let coefficient1 =
+            compress_ciphertext_coefficient(10, to_unsigned_representative(coefficients[0]));
```

Comment:
Couldn't `compress_ciphertext_coefficient` take an `i32` and call `to_unsigned_representative` itself? This would make the code less noisy because you don't need to call `to_unsigned_representative` so many times.
If you want to keep the preconditions on `compress_ciphertext_coefficient`, you could add one function in between that does the conversion.

Reply:
This could work, but I'd rather investigate it and put up code in a separate PR.